tests/TestCompiler.hs

```diff
@@ -197,27 +197,29 @@ runTest = P.runMake P.defaultOptions
 
 compile
   :: [(P.Module, P.ExternsFile)]
+  -> M.Map P.ModuleName FilePath
   -> [FilePath]
   -> ([P.Module] -> IO ())
   -> IO (Either P.MultipleErrors [P.ExternsFile], P.MultipleErrors)
-compile supportExterns inputFiles check = silence $ runTest $ do
+compile supportExterns supportForeigns inputFiles check = silence $ runTest $ do
   fs <- liftIO $ readInput inputFiles
   ms <- P.parseModulesFromFiles id fs
   foreigns <- inferForeignModules ms
   liftIO (check (map snd ms))
-  let actions = makeActions foreigns
+  let actions = makeActions (foreigns `M.union` supportForeigns)
```

Comment:
So here is where we `makeActions`.  This makes a `codegen` function in `Make.hs` which checks for an FFI file using the `foreigns` argument passed to `makeActions`.  But we were only passing in the foreign files for the modules of the test input files (I think).  So I've passed through the FFI support files too.  The tests started passing once I did this.